flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -304,8 +305,12 @@ public void start() throws Exception {
 					RpcUtils.getHostname(commonRpcService),
 					ConfigurationUtils.getSystemResourceMetricsProbingInterval(configuration));
 
+				final int poolSize = configuration.getInteger(ClusterOptions.CLUSTER_IO_EXECUTOR_POOL_SIZE, Hardware.getNumberCPUCores());
+				Preconditions.checkArgument(poolSize > 0,
+					String.format("Illegal pool size (%s) of io-executor for mini cluster, please re-configure '%s'.",
+						poolSize, ClusterOptions.CLUSTER_IO_EXECUTOR_POOL_SIZE.key()));
```

Comment:
The logic to obtain the `poolSize` and to verify it could be factored out into a method which is shared by the `MiniCluster` and the `ClusterEntrypoint`. This would avoid code duplication.